zer0-os/zOS
# fix: handle processing undecryptable message edits

### What does this do?
- handles the processing of undecryptable message edits by replacing the message with `Message edit cannot be decrypted.`.

### Why are we making this change?
- We currently ignore message edits that are undecryptable. The changes ensure that we display something else rather than show the original message / empty message.

### How do I test this?
- locate a message that has been edited and the `msgtype` is `m.bad.encrypted` > check the message displays: `Message edit cannot be decrypted`

### Key decisions and Risk Assessment:
  #### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

BEFORE:
<img width="686" alt="Screenshot 2023-11-27 at 12 34 25" src="https://github.com/zer0-os/zOS/assets/39112648/13360fb9-4e61-4896-af7b-49ffe090ace5">

<img width="686" alt="Screenshot 2023-11-27 at 12 35 06" src="https://github.com/zer0-os/zOS/assets/39112648/364974a2-32f5-41cf-911e-4547226fd925">

AFTER:

<img width="686" alt="Screenshot 2023-11-27 at 12 33 10" src="https://github.com/zer0-os/zOS/assets/39112648/8a3d3959-af5b-4ddb-bb9b-8d563d808a4d">

<img width="686" alt="Screenshot 2023-11-27 at 12 35 39" src="https://github.com/zer0-os/zOS/assets/39112648/6c12cc71-c280-4b9c-a43b-020e7486458f">
